website/docs/reference/warehouse-setups/snowflake-setup.md

```diff
@@ -124,6 +124,8 @@ Along with adding the `authenticator` parameter, be sure to run `alter account s
 
 To use key pair authentication, omit a `password` and instead provide a `private_key_path` and, optionally, a `private_key_passphrase` in your target. **Note:** Versions of dbt before 0.16.0 required that private keys were encrypted and a `private_key_passphrase` was provided. This behavior was changed in dbt v0.16.0.
 
+From dbt v1.5.0 onwards, it is possible to provide a `private_key` (Base64-encoded DER-encoded key bytes) string instead of a `private_key_path`.
```

Comment:
@b-per @dataders do we want to just version this section so it appears for 1.5 and higher? also do we prefer users to use the `private_key` instead of `private_key_path`?

I'm happy to make that change on my end, but I'm wondering if it's a better method so that the content doesn't go stale in the future

Reply:
We can make it dependent on the version, but it will just require a bit more changes (and I added an example in the code block below as well)

There is no preferred way between `private_key` and `private_key_path`, but I have heard some people who wanted to use `private_key` because they were calling Snowflake as part of CI/CD and didn't want to copy the key to a file as it was against their security policies. Using `private_key` with a CI secret was OK with them.